src/main/java/org/traccar/protocol/TopinProtocolDecoder.java

```diff
@@ -60,13 +63,28 @@ private void sendResponse(Channel channel, int length, int type, ByteBuf content
         }
     }
 
+    private void updateTime(Channel channel, int type) {
+        ByteBuf dateBuffer = Unpooled.buffer();
+
+        Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+
+        dateBuffer.writeShort(calendar.get(Calendar.YEAR));
+        dateBuffer.writeByte(calendar.get(Calendar.MONTH)+1);
+        dateBuffer.writeByte(calendar.get(Calendar.DAY_OF_MONTH));
+        dateBuffer.writeByte(calendar.get(Calendar.HOUR_OF_DAY));
+        dateBuffer.writeByte(calendar.get(Calendar.MINUTE));
+        dateBuffer.writeByte(calendar.get(Calendar.SECOND));
+
+        sendResponse(channel, dateBuffer.readableBytes(), type, dateBuffer);
+    }
+
     @Override
     protected Object decode(
             Channel channel, SocketAddress remoteAddress, Object msg) throws Exception {
 
         ByteBuf buf = (ByteBuf) msg;
 
-        buf.skipBytes(2); // header
```

Comment:
Please leave the comments that were already in the code.

Reply:
Done! c990bf7